### DISCUSSION TITLE
Definir padrão de commit

### DISCUSSION THREAD

**Discussion by CassianoJunior** [diego3g/rsxp-2023#26] [Ideas]
Seria interessante usar as bibliotecas que o @diego3g mostrou no [Code/Drops #12](https://www.youtube.com/watch?v=erInHkjxkL8) para padronização dos commits?
- [Husky](https://github.com/typicode/husky)
- [CommitLint](https://github.com/conventional-changelog/commitlint)
- [Commitizen](https://github.com/commitizen/cz-cli)

**Comment by Rodrigo001-dev** [comment]
Gostei da sugestão, com isso podemos fazer todos os colaboradores seguirem o mesmo padão de commit.

Nessa discussão foi sugerido em utilizar padronização de Pull Request da Brasil API  #18
Podemos juntar as duas ideias.

**Comment by henriqgoncalvs** [comment]
Pra usar essas ferramentes a gente precisaria fazer uma configuração de workspace. Pq a gente precisa usar o `.git` do projeto, daí n daria pra instalar o husky separadamente em cada app.

Por mim é tranquilo fazer essa configuração simples, peguei até algumas referências pra ajudar:

https://daveiscoding.com/nodejs-typescript-monorepo-via-npm-workspaces
https://plainenglish.io/blog/monorepo-setup-with-npm-and-typescript 

Daí a gente consegue criar 2 scripts diferentes (1 pra mobile e outro pra server) e roda os dois sempre que for fazer algum commit.

https://dev.to/monfernape/enforce-husky-pre-commit-with-eslint-prettier-in-monorepo-55jc

@diego3g se puder dar uma iluminada aqui, n tenho tanta experiência com monorepo então posso tá falando besteira kk

**Comment by henriqgoncalvs** [comment]
Sobre o commitizen, acho que não é muito necessário. Só o [commitlint](https://github.com/conventional-changelog/commitlint) já seria massa.

Mas aí fica a cargo de todo mundo decidir

**Reply by goncadanilo** [reply]
A vantagem do commitizen é que ele traz as opções do tipo de commit, talvez para quem não tem familiaridade com esse padrão de commit seria legal usar. Mas fica a critério do que o pessoal decidir.

**Comment by diego3g** [comment]
Concordo com o @hnqg, acho que podemos simplificar agora e trazer apenas uma PR que padroniza os commits de maneira menos burocrática possível. Fora isso, não acho legal colocarmos muito empecilho para as pessoas contribuírem então quando menos ferramentas melhor.

Se alguém for pegar pra adicionar o `commitlint` peço que criem uma issue e sinalizem que estão trabalhando nisso.

Vou fechar essa discussão porque estamos com bastante discussões abertas com tópicos que super valem a contribuição de vocês por lá também e acho que não vale extender muito a discussão por aqui. Sintam-se livres pra reabrir a discussão se fizer sentido 🤗
